library/src/main/java/com/datatorrent/lib/fileaccess/FileAccess.java

```diff
@@ -151,7 +151,7 @@
      * @param value
      * @throws IOException
      */
-    void append(byte[] key, byte[] value) throws IOException;
+    void append(Slice key, Slice value) throws IOException;
```

Comment:
@davidyan74 
Adding a method to an interface is a binary compatible change. 
https://docs.oracle.com/javase/specs/jls/se7/html/jls-13.html